modules/hitl2/src/backend/repository.ts

```diff
@@ -0,0 +1,367 @@
+  private recentConversationQuery(): Knex.QueryBuilder {
+    return this.bp
+      .database('events')
+      .select('*')
+      .whereIn('id', function() {
+        this.max('id')
+          .from('events')
+          .groupBy('threadId')
+      })
+      .as('most_recent_event')
+  }
+
+  private userEventsQuery() {
+    return this.bp
+      .database<EscalationType>('escalations')
+      .select(
+        'escalations.id as escalation:id',
+        'escalations.userThreadId as escalation:userThreadId',
+        'most_recent_event.*'
+      )
+      .join(
+        this.recentConversationQuery().where('direction', 'incoming'),
+        'escalations.userThreadId',
+        'most_recent_event.threadId'
+      )
+  }
+
+  private escalationsWithCommentsQuery(botId: string, conditions: CollectionConditions = {}): Knex.QueryBuilder {
+    const { limit, orderByColumn, orderByDirection } = conditions
+
+    return this.bp
+      .database('escalations')
+      .select(
+        'escalations.*',
+        `comments.id as ${this.commentPrefix}:id`,
+        `comments.agentId as ${this.commentPrefix}:agentId`,
+        `comments.content as ${this.commentPrefix}:content`,
+        `comments.updatedAt as ${this.commentPrefix}:updatedAt`,
+        `comments.createdAt as ${this.commentPrefix}:createdAt`
+      )
+      .leftJoin('comments', 'escalations.id', 'comments.escalationId')
+      .andWhere('escalations.botId', botId)
+      .distinct()
+      .modify(this.applyLimit, limit)
+      .modify(this.applyOrderBy, orderByColumn, orderByDirection)
+      .orderBy([{ column: 'comments.createdAt', order: 'asc' }])
+  }
+
+  private applyQuery = (query?: Knex.QueryCallback) => {
+    return (builder: Knex.QueryBuilder) => {
+      if (query) {
+        return builder.modify(query)
+      } else {
+        return builder
+      }
+    }
+  }
+
+  escalationsQuery = async (query?: Knex.QueryCallback): Promise<EscalationType[]> => {
+    return await this.bp.database<EscalationType>('escalations').modify(this.applyQuery(query))
+  }
+
+  getAgentOnline = async (botId: string, agentId: string): Promise<boolean> => {
+    const value = await this.bp.kvs.forBot(botId).get(`hitl2:online:${agentId}`)
+    return !!value
+  }
+
+  setAgentOnline = async (botId: string, agentId: string, value: boolean): Promise<boolean> => {
+    const config = await this.bp.config.getModuleConfigForBot('hitl2', botId)
+    await this.bp.kvs.forBot(botId).set(`hitl2:online:${agentId}`, value, null, config.agentSessionTimeout)
+    return value
+  }
+
+  getCurrentAgent = async (req, botId: string, agentId: string): Promise<AgentType> => {
+    const online = await this.getAgentOnline(botId, agentId)
+
+    return axios
+      .get('/auth/me/profile', await this.axiosConfig(req, botId))
+      .then(response => response.data.payload)
+      .then(data => {
+        return {
+          ...data,
+          id: agentId,
+          online: online
+        } as AgentType
+      })
```

Comment:
Please use await instead of chaining then. It's a lot clearer that way
````js
const {data} = await axios.get('/auth/me/profile', await this.axiosConfig(req, botId))
 
return {
  ...data.payload,
  id: agentId,
  online: online
} as AgentType
```